pkg/openshift/util/console_links.go

```diff
@@ -0,0 +1,112 @@
+package util
+
+import (
+	"encoding/json"
+	"errors"
+
+	hawtiov1alpha1 "github.com/hawtio/hawtio-operator/pkg/apis/hawtio/v1alpha1"
+
+	consolev1 "github.com/openshift/api/console/v1"
+	routev1 "github.com/openshift/api/route/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ConsoleLinkConfig defines the properties required to create a console link
+type ConsoleLinkConfig struct {
```

Comment:
Now that we have a `HawtConfig` object, I'd suggest we remove that intermediate abstraction.

Reply:
ConsoleLinkConfig has more data than what's available in hawtconfig.json. Do you prefer to pass the additional data as separate parameters to functions?

Reply:
Yes, that's mainly the route, so better passing as an extra parameter than creating an extra struct just for that.

Reply:
Done